wgpu-hal/src/vulkan/command.rs

```diff
@@ -52,6 +51,51 @@ impl super::CommandEncoder {
             }
         }
     }
+
+    fn make_framebuffer(
+        &mut self,
+        key: super::FramebufferKey,
+    ) -> Result<vk::Framebuffer, crate::DeviceError> {
+        Ok(match self.framebuffers.entry(key) {
+            Entry::Occupied(e) => *e.get(),
+            Entry::Vacant(e) => {
+                let vk_info = vk::FramebufferCreateInfo::default()
+                    .render_pass(e.key().raw_pass)
+                    .width(e.key().extent.width)
+                    .height(e.key().extent.height)
+                    .layers(e.key().extent.depth_or_array_layers)
+                    .attachments(&e.key().attachments);
+
+                let raw = unsafe { self.device.raw.create_framebuffer(&vk_info, None).unwrap() };
+                *e.insert(raw)
```

Comment:
thought: Sigh, I wish nice helpers like `Entry::or_insert_with` didn't have to be sacrificed when we need to bail out with `?` if something goes wrong.